### PR TITLE
update project phase from most recent published docs

### PR DESCRIPTION
**Pull request by mtCarto**
- updates project phase based on EA Decision and most recently published document's phase
 - adds a phaseHistory field to the project model for future updates by document interactions

 - updates the api search controller to unwind the phase field
 - updates models and factories

### CODE REVIEW

**Comment by mtCarto** [comment]
We should wait to merge this and the associated public and admin PR's, until we have confirmed the accuracy of the phase tagging on projects.

File: migrations/20191230174852-projPhaseFromDocs.js [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+// pass project
+function matchPhaseToDecision(decisionPhases, docPhaseStack, phases) {
+  let phaseToTag = '';
+  let previousPhases = [];
+  let stackCopy = [...docPhaseStack];
+
+  if (docPhaseStack.length === 1) {
+    phaseToTag = docPhaseStack.pop()
+  }
+  // todo is this getting hit?
```
Comment (KitArmstrong) [inline migrations/20191230174852-projPhaseFromDocs.js]: Did this run?

File: migrations/20191230174852-projPhaseFromDocs.js [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+
+  var phaseToTagId = lookupId(phases, phaseToTag);
+  var previousPhasesIds = namesToIds(phases, previousPhases);
+  return { "currentPhase": phaseToTagId, "previousPhases": previousPhasesIds, "previousNames": previousPhases }
+}
+
+function lookupId(phases, name) {
+  // console.log("looking up: ", name);
+  var phaseId;
+  [phaseId] = phases.filter(phase => {
```
Comment (KitArmstrong) [inline migrations/20191230174852-projPhaseFromDocs.js]: Not sure that the square brackets are required. They generally indicate computed properties, but shouldn't be required here.

Comment (mtCarto) [reply migrations/20191230174852-projPhaseFromDocs.js]: Just some destructuring

**Pull request merged**